File: Typescript/test/reversePrefix.test.ts
import { expect, test } from '@jest/globals';
import * as solution from '../src/reversePrefix';

test('Example 1', () => {
	let input0 = "abcd";
	let input1 = 2;
	let expected = "bacd";
	expect(solution.reversePrefix(input0, input1)).toEqual(expected);
});

test('Example 2', () => {
	let input0 = "xyz";
	let input1 = 3;
	let expected = "zyx";
	expect(solution.reversePrefix(input0, input1)).toEqual(expected);
});

test('Example 3', () => {
	let input0 = "hey";
	let input1 = 1;
	let expected = "hey";
	expect(solution.reversePrefix(input0, input1)).toEqual(expected);
});

File: Dart/test/reversePrefix_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:leetcode/reversePrefix.dart';

void main() {
  test("test example 1", () {
    var input0 = "abcd";
    var input = 2;
    var expected = "bacd";
    final sln = Solution();
    var output = sln.reversePrefix(input0, input);
    expect(output, expected);
  });
  test("test example 2", () {
    var input0 = "xyz";
    var input = 3;
    var expected = "zyx";
    final sln = Solution();
    var output = sln.reversePrefix(input0, input);
    expect(output, expected);
  });
  test("test example 3", () {
    var input0 = "hey";
    var input = 1;
    var expected = "hey";
    final sln = Solution();
    var output = sln.reversePrefix(input0, input);
    expect(output, expected);
  });
}
